### ISSUE TITLE
"Zero wind" (`00000KT`) scenario not completely handled

### ISSUE THREAD

**Issue by SkySails** [mivek/python-metar-taf-parser#61]
Hello! First of all, thank you for your awesome work on this family of parsers 🙌 

I am debugging an issue in the [Typescript port of this projec](https://github.com/aeharding/metar-taf-parser)t, and it seems like the same thing occurs over here and since that other project is a port, I think it's best to work with this upstream first!
The following METAR report incorrectly parses to the wrong unit:

```
KATL 270200Z 00000MPS
```

The result, as per https://www.metar-taf-decoder.com, is `Wind: Speed : 0 KT Direction : North | 0°` - `KT` instead of `MPS`.
This is a really small issue, but it has other consequences in the Typescript port which you can read about in the linked issues at the bottom!

### Related issues

- https://github.com/aeharding/metar-taf-parser/issues/80
- https://github.com/aeharding/metar-taf-parser/issues/79

### More details

The culprit seems to be this regex, as it results in the following incorrect matching:

https://github.com/mivek/python-metar-taf-parser/blob/164900fcf0dd57276a9c0609c37f423c628e1acf/metar_taf_parser/command/common.py#L82

It seems to be there to satisfy [this test case](https://github.com/mivek/python-metar-taf-parser/blob/164900fcf0dd57276a9c0609c37f423c628e1acf/metar_taf_parser/tests/parser/test_parser.py#L293), which includes an invalid METAR string according to the specifications I can find. As usual, most are not too explicit on the matter but this is one example:

>Wind speeds (ff) are two digits (or three digits if required), in knots.
><small>https://meteocentre.com/doc/metar.html</small>

So I guess this is a case of choosing which case to support. If we change the regex by adding an additional required zero:

```regex
(VRB|000|[0-3]\d{2})(\d{2})G?(\d{2,3})?(KT|MPS|KM\/H)?
```

The unit is now correctly parsed and the speed is correctly registered (not that it matters, it's all zeros anyway)

![image](https://github.com/aeharding/metar-taf-parser/assets/46646495/861edb1a-ea24-49f0-bd0b-4cd5bf4ee28c)

The downside is that the test case will start to fail. Is there a reason behind supporting `0000KT` (four digits) specifically? Is it a common occurrence in the real world?

**Comment by mivek** [comment]
Hello @SkySails 

Sorry for the delay.
The support for the `0000KT` token actually comes from a PR from the Typescript project:
 - https://github.com/aeharding/metar-taf-parser/pull/66
The original METAR was `KATW 022045Z 0000KT 10SM SCT120 00/M08 A2996` however after checking in the [Federal Meteorological Handbook](https://www.icams-portal.gov/resources/ofcm/fmh/FMH1/fmh1_2019.pdf), it seems that `0000KT` is invalid.
Here is an extract of section 12.6.5 regarding the Calm wind

> d) Calm Wind. Calm wind shall be coded as "00000KT"
The first three zeros represent the wind direction and the last two zeros represent the speed.

```
(VRB|000|[0-3]\d{2})(\d{2})G?(\d{2,3})?(KT|MPS|KM\/H)?
```
is the right regex.

Thank you for your analysis of the bug, I will push a fix.

**Comment by SkySails** [comment]
No worries at all! 

Ah, I should have looked through the issue history too, that's my bad. Thank you for referencing it!
You're welcome, I'm happy to help!

**Comment by mivek** [comment]
Hello @SkySails 

I merged the fix and released a new version `1.8.2`. I will report the fix in the Java version in the next few days.

Thanks again for reporting the issue

**Comment by SkySails** [comment]
That's great news! Thank _you_ for fixing it!
